app/jobs/submission_invoice_creation_job.rb

```
@@ -1,5 +1,8 @@
 class SubmissionInvoiceCreationJob < ApplicationJob
   def perform(submission)
-    Workday::SubmitCustomerInvoiceRequest.new(submission).perform
+    return if submission.invoice.present?
```

I do wonder if it's better that we raise an exception here rather than silently carrying on when an attempt is made to create an invoice against a submission that already has one. In theory this should never happen so if it does, knowing about it would be useful so we know that something is going wrong somewhere.

---

I considered that too, but couldn't think of what error. Just a StandardError? Make a new one?

---

I think the default `RuntimeError` raised by default would be fine: `raise "Submission #{submission.id} already has an invoice!"`